### PR TITLE
Add ifrc presence items

### PR DESCRIPTION
**Pull request by barshathakuri**
## Addresses:
- https://github.com/toggle-corp/ifrc-go-meta/issues/372

## Changes
- Add ifrc and icrc presence item

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/views/CountryProfileSupportingPartners/PresenceItem/index.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+function PresenceItem() {
+    const strings = useTranslation(i18n);
+
+    const { countryResponse } = useOutletContext<CountryOutletContext>();
+    const icrcPresenceDetail = countryResponse?.icrc_presence;
+
+    const legalStatusLink = 'https://idp.ifrc.org/SSO/SAMLLogin?loginToSp=https://fednet.ifrc.org&returnUrl=https://fednet.ifrc.org/PageFiles/255835/List%20States%20with%20Defined%20Legal%20Status%2025.07.2023_ENG.pdf';
```
Comment (samshara) [inline src/views/CountryProfileSupportingPartners/PresenceItem/index.tsx]: Let's move it outside of component

File: src/views/CountryProfileSupportingPartners/PresenceItem/index.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            <Container
+                className={styles.presenceCard}
+                childrenContainerClassName={styles.presenceCardList}
+                heading={strings.countryIFRCPresenceTitle}
+            >
+                <div className={styles.ifrcPresenceItem}>
+                    <TextOutput
+                        label={strings.countryIFRCPresenceHeadOfDelegation}
+                        value={undefined}
+                    />
```
Comment (samshara) [inline src/views/CountryProfileSupportingPartners/PresenceItem/index.tsx]: Let's add a TODO here

File: src/views/CountryProfileSupportingPartners/PresenceItem/index.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                        withUnderline
+                    >
+                        {strings.countryIFRCLegalStatus}
+                    </Link>
+                </div>
+                <div className={styles.ifrcPresenceItem}>
+                    <TextOutput
+                        label={strings.countryIFRCContact}
+                        value={undefined}
+                    />
```
Comment (samshara) [inline src/views/CountryProfileSupportingPartners/PresenceItem/index.tsx]: Let's add a TODO

File: src/views/CountryProfileSupportingPartners/PresenceItem/index.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                        {strings.countryIFRCDisasterLaw}
+                    </Link>
+                </div>
+            </Container>
+            {icrcPresenceDetail?.map((icrc) => (
+                <Container
+                    className={styles.presenceCard}
+                    heading={strings.countryICRCPresenceTitle}
+                    childrenContainerClassName={styles.presenceCardList}
+                >
```
Comment (samshara) [inline src/views/CountryProfileSupportingPartners/PresenceItem/index.tsx]: Let's use `key` here

File: src/views/CountryProfileSupportingPartners/PresenceItem/index.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+import { resolveToString } from '#utils/translation';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+function PresenceItem() {
+    const strings = useTranslation(i18n);
+
+    const { countryResponse } = useOutletContext<CountryOutletContext>();
+    const icrcPresenceDetail = countryResponse?.icrc_presence;
```
Comment (samshara) [inline src/views/CountryProfileSupportingPartners/PresenceItem/index.tsx]: Let's confirm if `icrcPresenceDetail` is an array field?

File: src/views/CountryProfileSupportingPartners/PresenceItem/index.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+                <Container
+                    className={styles.presenceCard}
+                    heading={strings.countryICRCPresenceTitle}
+                    childrenContainerClassName={styles.presenceCardList}
+                >
+                    {resolveToString(
+                        strings.countryICRCConfirmedPartner,
+                        { year: getCurrentMonthYear() },
+                    )}
+                    <div className={styles.icrcPresenceItem}>
```
Comment (samshara) [inline src/views/CountryProfileSupportingPartners/PresenceItem/index.tsx]: we should only show this text if the `icrc_presence` is `true`. Let's remove the whole section if it's `false`.

**Review by roshni73 — APPROVED** [review]

**Review by samshara — APPROVED** [review]

**Pull request merged**